client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -364,11 +364,22 @@ const Editor = (props: EditorProps): JSX.Element => {
     editor.updateOptions({
       accessibilitySupport: accessibilityMode ? 'on' : 'auto'
     });
-    // Users who are using screen readers should not have to move focus from
-    // the editor to the description every time they open a challenge.
-    if (props.canFocus && !accessibilityMode) {
-      focusIfTargetEditor();
-    } else focusOnHotkeys();
+
+    const isTablistShowing = (): boolean =>
+      !!document.querySelector('.nav-tabs');
+    const isUsingKeyboard = (): boolean =>
+      document.querySelector('.nav-tabs')?.getAttribute('data-keyboard') ===
+      'true';
```

Comment:
Can we just pass props down to the editor?  I don't think we need to inspect the DOM here.

`isTablistShowing` is always true if we're using the mobile layout and `isUsingKeyboard` could be state of `show.tsx` since it has both the editor and the layout as children.

Reply:
@ojeytonwilliams I hope these changes are a little more indicative of the Reacty way. Thanks for steering me in the right direction.